### ISSUE TITLE
Support specifying an AWS Role to access cross-account resources

### ISSUE THREAD

**Issue by beanaroo** [ToolJet/ToolJet#1738]
### Is your proposal related to a problem?

In a multi-account structure such as an AWS Organization, shared services like ToolJet may be hosted in a different account to where application resources are hosted.

Accessing different accounts using a single set of credentials is done by assuming relevant AWS Roles to issue temporary credentials.

### Describe the solution you'd like

Introduce a field when configuring a DynamoDB or S3 datasource to specify a Role ARN.

### Describe alternatives you've considered

None

### If the feature is approved, would you be willing to submit a PR?

I would be willing to attempt some changes to the server-side. For DynamoDB, I believe the relevant section is: https://github.com/ToolJet/ToolJet/blob/26c9cc655c9774936e1007c07263cac89cdf181d/server/plugins/datasources/dynamodb/index.ts#L52-L53

Basically, if `sourceOptions['role_arn']` is provided, use the existing `credentials` object with [ChainableTemporaryCredentials](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/ChainableTemporaryCredentials.html) to get a new `credentials` object.

I am not confident enough to make any frontend changes.

This improvement would compliment, but does not depend on, https://github.com/ToolJet/ToolJet/issues/1737

**Comment by akshaysasidrn** [comment]
Similar to #1737, changes at frontend will need to be done only on the schema file. If you are picking to implement this, I can help you out with that if there are any hurdles.
Else would need some time to pick it up to test this out on an ec2 instance.
